Add homematic state_uncertain attribute for the Keylock device

trilu2000 opened the pull request:
## Description:
The Homematic Keylock Device holds an attribute to show if the status is defined or the lock was opened manually. This flag is called - STATE_UNCERTAIN
To make this attribute working, it is necassary to add the flag to the dictionary, which i have done.

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [ ] There is no commented out code in this PR.

trilu2000 committed:
Added state_uncertain attribute to the dictionary for the Keylock device

homeassistant commented:
Hi @trilu2000, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22660).

Once you do that we will be able to review and accept this pull request.

Thanks!

trilu2000 committed:
added attribute transition to homematic light devices

light.turn_on provides an attribute transition which gives in combination with a brightness value a nice dimming effect

danielperna84 commented on homeassistant/components/homematic/light.py:
```diff
@@ -77,6 +77,9 @@ def effect(self):
     def turn_on(self, **kwargs):
         """Turn the light on and/or change color or color effect settings."""
+        if ATTR_TRANSITION in kwargs:
+            self._hmdevice.setValue('RAMP_TIME', kwargs[ATTR_TRANSITION])
```
Apart from making `RAMP_TIME` a variable (`ATTR_RAMP_TIME = 'RAMP_TIME'` somewhere at the top) for the sake of consistency within HA everything seems fine to me. 👍

danielperna84 commented:
Oh, and of course the tests have to succeed.

```
./homeassistant/components/homematic/light.py:5:80: E501 line too long (85 > 79 characters)
./homeassistant/components/homematic/light.py:82:1: W293 blank line contains whitespace
```
Those are the two errors.

MartinHjelmare commented:
There hasn't been any movement here for more than a month. I'll close now. Please open a new PR when ready to finish.
